app/src/test/java/com/android/sample/receivers/NotificationReceiverUnitTest.kt

```diff
@@ -57,4 +58,39 @@ class NotificationReceiverUnitTest {
       "Notification content does not match"
     }
   }
+
+  @Test
+  fun `onReceive should create deletion notification with correct content`() {
+    val activityId = "123"
+    val activityName = "Test Activity"
+    val notificationTitle = "Activity Deleted"
+
+    val intent =
+        Intent().apply {
+          putExtra("activityId", activityId)
+          putExtra("activityName", activityName)
+          putExtra("notificationTitle", notificationTitle)
+          putExtra("isDeletionNotification", true)
+          putExtra("userId", "u1")
+        }
+
+    notificationReceiver.onReceive(context, intent)
+
+    // Verify that a notification was posted
+    val notifications = shadowNotificationManager.allNotifications
+    assert(notifications.isNotEmpty()) { "No notifications were posted" }
+
+    val notification = notifications[0]
+    val shadowNotification = Shadows.shadowOf(notification)
+
+    assert(shadowNotification.contentTitle == notificationTitle) {
+      "Notification title does not match"
+    }
+    assert(shadowNotification.contentText == "Join or create another activity!") {
+      "Deletion notification content does not match"
+    }
+    assert(notification.category == NotificationCompat.CATEGORY_EVENT) {
+      "Notification category should be EVENT for deletion notifications"
+    }
+  }
 }
```

Comment:
Good job for your current tests for the helper and receiver and the 100% coberage achieved on them. There are no current tests for ScheduleNotification and sendDeletionNotification that are in the app file, you could maybe add test for them or for the code you added in the listActivityViewModel when an activity is deleted to reach 80% of coverage on new code.  Also you should use users from dummy data rather than creating new users each time for your list of participants.

Reply:
In think the test for App.kt has to be handled with the end-to-end tests as it contains app initialization. Concerning the other tests, we will be refactoring all tests before deadline always. Right now I have a technical issue and unfortunately can't add a test in listActivityViewModel tests

Reply:
ok i see, this is okay then